web/files/files.go

```diff
@@ -293,6 +293,75 @@ func UploadMetadataHandler(c echo.Context) error {
 	return jsonapi.Data(c, http.StatusCreated, &m, nil)
 }
 
+// FileCopyHandler handles POST requests on /files/:file-id/copy
+//
+// It is used to duplicate the given file and its metadata except for
+// relationships.
+func FileCopyHandler(c echo.Context) error {
+	instance := middlewares.GetInstance(c)
+	fs := instance.VFS()
+
+	fileID := c.Param("file-id")
+	olddoc, err := instance.VFS().FileByID(fileID)
+	if err != nil {
+		return WrapVfsError(err)
+	}
+
+	err = checkPerm(c, permission.GET, nil, olddoc)
+	if err != nil {
+		return err
+	}
+
+	// TODO: move block to separate method?
+	newdoc := olddoc.Clone().(*vfs.FileDoc)
+	newdoc.DocID = ""
+	newdoc.DocRev = ""
+	newdoc.DocName = olddoc.DocName + " (copy)"
+	newdoc.Metadata.RemoveCertifiedMetadata()
+	newdoc.CozyMetadata = nil
+	newdoc.ReferencedBy = nil
+	newdoc.InternalID = ""
+	newdoc.CreatedAt = time.Now()
+	newdoc.UpdatedAt = time.Now()
+	newdoc.RestorePath = ""
```

Comment:
I don't see what you want to do here. We have two cases:

- the file wasn't in the trash, and `RestorePath` is already blank
- the file was in the trash, and we should either block the action or keep the `RestorePath`

Reply:
The idea was to regenerate the restore path based on the directory and new name. I must have forgotten about it. I'm gonna fix thsi.

Reply:
Ah I thought `RestorePath` contained the name of the document but it's only its old parent path. Since the parent of the copy should be the same as the original one we don't need to do anything here.